src/sql/workbench/contrib/notebook/browser/notebookViews/notebookViewsGrid.component.ts

```diff
@@ -0,0 +1,137 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import 'vs/css!./notebookviews';
+import { Component, OnInit, ViewChildren, QueryList, Input, Inject, forwardRef, ChangeDetectorRef } from '@angular/core';
+import { NotebookViewsCardComponent } from 'sql/workbench/contrib/notebook/browser/notebookViews/notebookViewsCard.component';
+import { ICellModel } from 'sql/workbench/services/notebook/browser/models/modelInterfaces';
+import { NotebookModel } from 'sql/workbench/services/notebook/browser/models/notebookModel';
+import { GridStack } from 'gridstack';
+import 'gridstack/dist/h5/gridstack-dd-native';
+import { IColorTheme, ICssStyleCollector, registerThemingParticipant } from 'vs/platform/theme/common/themeService';
+import { localize } from 'vs/nls';
+import { NotebookViewsExtension } from 'sql/workbench/services/notebook/browser/notebookViews/notebookViewsExtension';
+import { CellChangeEvent, INotebookViewCell } from 'sql/workbench/services/notebook/browser/notebookViews/notebookViews';
+
+@Component({
+	selector: 'notebook-views-grid-component',
+	templateUrl: decodeURI(require.toUrl('./notebookViewsGrid.component.html'))
+})
+export class NotebookViewsGridComponent implements OnInit {
+	@Input() cells: ICellModel[];
+	@Input() model: NotebookModel;
+	@Input() views: NotebookViewsExtension;
+
+	@ViewChildren(NotebookViewsCardComponent) private _items: QueryList<NotebookViewsCardComponent>;
+
+	protected _grid: any;
+	public loaded: boolean;
+
+	constructor(
+		@Inject(forwardRef(() => ChangeDetectorRef)) private _changeRef: ChangeDetectorRef,
+	) {
+		this.loaded = false;
+	}
+
+	public get empty(): boolean {
+		return !this._items || !this._items.find(item => item.display);
+	}
+
+	public get hiddenItems(): NotebookViewsCardComponent[] {
+		return this._items.filter(item => !item.display);
+	}
+
+	public get emptyText(): String {
+		return localize('emptyText', "This view is empty. Add a cell to this view by clicking the Insert Cells button.");
+	}
+
+	ngOnInit() {
+		const self = this;
+
+		setTimeout(() => {
+			self._grid = GridStack.init({
+				alwaysShowResizeHandle: false,
+				styleInHead: true
+			});
+
+			this.cells.forEach((cell) => self._grid);
+
+			this.loaded = true;
+			this.detectChanges();
+
+			self._grid.on('added', function (e, items) { self.persist('added', items, self._grid, self._items); });
```

Comment:
Can you give these parameters proper typings?

Reply:
Done.